boards/adafruit/itsybitsy/Kconfig

```diff
@@ -3,7 +3,7 @@
 # Copyright (c) 2022 Embla Flatlandsmo <embla.flatlandsmo@gmail.com>
 # SPDX-License-Identifier: Apache-2.0
 
-config BOARD_SERIAL_BACKEND_CDC_ACM
-	bool "USB CDC"
+config BOARD_NEED_SERIAL_BACKEND_CDC_ACM
```

Comment:
No. This is used in the board context and will stay prefix with `BOARD`. `NEED` is already used in Kconfig files.

Reply:
> This is used in the board context and will stay prefix with BOARD.

yes in your PR, and my point is that this could potentially be useful in other contexts in the future.
Having `BOARD` as prefix in this case give no added value compared to removing `BOARD`.

> `NEED` is already used in Kconfig files.

I didn't say `NEED` is not used, but I said `REQUIRES` is more used:
> > `REQUIRES` is slightly more used term in Kconfig than `NEED`

Trying to be consistent in Zephyr is important, and in this case trying to align the most used naming is better than on purpose making things different on purpose.

Reply:
> yes in your PR, and my point is that this could potentially be useful in other contexts in the future.
> Having `BOARD` as prefix in this case give no added value compared to removing `BOARD`.

No, there actually no future, for all use cases we have cdc-acm-console snippet, it just that hello_world should work "out of the box" without any additional command line arguments.

>Trying to be consistent in Zephyr is important, and in this case trying to align the most used naming is better than on purpose making things different on purpose.

Yes, `git grep "config .*_NEED_"` vs `git grep "config .*_REQUIRED_"`

Reply:
> No, there actually no future, for all use cases we have cdc-acm-console snippet, 

and you now all the directions the future will take ?
What about the day when it's the SoC or a CPU cluster which will be requiring a specific backend.
cdc acm as serial backend, or a CPU cluster which will requires an IPC serial backend ?

I personally don't know the future, hence I believe it better to avoid the `BOARD` prefix as it doesn't provide any added benefit imo.

> > Trying to be consistent in Zephyr is important, and in this case trying to align the most used naming is better than on purpose making things different on purpose.
> 
> Yes, `git grep "config .*_NEED_"` vs `git grep "config .*_REQUIRED_"`

I said `REQUIRES`, not `REQUIRED`, not to mention that your grep of using `_` in front and after the keyword limits the number of hits. 
The majority use of `NEED` is in the net subsystem, not on the highlevel, where, as example, the `REQUIRES_FULL_LIBC` exists.

Reply:
We should be using common patterns in tree not having each area use a different terminology, there's different terms used in different things like specifications i.e. "shall have" vs "must have", requires aligns more with that in specifications that I have read, I don't think I've seen any that use need

Reply:
> > No, there actually no future, for all use cases we have cdc-acm-console snippet,
> 
> and you now all the directions the future will take ? What about the day when it's the SoC or a CPU cluster which will be requiring a specific backend. cdc acm as serial backend, or a CPU cluster which will requires an IPC serial backend ?
> 
> I personally don't know the future, hence I believe it better to avoid the `BOARD` prefix as it doesn't provide any added benefit imo.

The whole purpose of the first commit is that boards do not want to use the cdc-acm-console snippet because it would require additional build parameters. And CDC ACM UART backend is not selected by this kconfig file, but by the chosen property, Kconfig file is needed because of USB device stack and few other settings for convenience.

> I said `REQUIRES`, not `REQUIRED`, not to mention that your grep of using `_` in front and after the keyword limits the number of hits. The majority use of `NEED` is in the net subsystem, not on the highlevel, where, as example, the `REQUIRES_FULL_LIBC` exists.

okay, changed to BOARD_REQUIRES_SERIAL_BACKEND_CDC_ACM

Reply:
If it would be acceptable [Kconfig.cdc_acm_serial.defconfig](https://github.com/zephyrproject-rtos/zephyr/pull/81308/commits/f2b55b00f36484e9452e2f5481346898a51e4563#diff-044af1612b89d9e03c0c285d11e2e5946a1c97bc2bc6a47a0349b3b8e52019c4) to be placed under the `subsys/usb` then I am fine with `REQUIRES_SERIAL_BACKEND_CDC_ACM`, also it could lead to a better integration.

Reply:
> If it would be acceptable to put [Kconfig.cdc_acm_serial.defconfig](https://github.com/zephyrproject-rtos/zephyr/pull/81308/commits/f2b55b00f36484e9452e2f5481346898a51e4563#diff-044af1612b89d9e03c0c285d11e2e5946a1c97bc2bc6a47a0349b3b8e52019c4) under the `subsys/usb` then I am fine with `REQUIRES_SERIAL_BACKEND_CDC_ACM`, also it might lead to a better integration.

why would that file need to be placed under `subsys/usb` ?
The symbols are defined here: https://github.com/zephyrproject-rtos/zephyr/blob/21bf9253163d565a92d52bfc06d3ad55b4b20bab/boards/Kconfig#L131
https://github.com/zephyrproject-rtos/zephyr/blob/f2b55b00f36484e9452e2f5481346898a51e4563/boards/Kconfig#L137
and those symbols could be moved to `subsys/usb`.

I'm not sure starting to allow `defconfig` files under various subsys / lib / etc. folders is a wise idea.
That will make it even harder to follow from where defconfig files are picked up.

And although defconfig file do adjust many settings defined throughout Zephyr repo, then I also understand the wish to have the definition close to the defconfigs impacted by this setting.
So I also see reasons to keep the symbol definition at current location.

Reply:
> > If it would be acceptable to put [Kconfig.cdc_acm_serial.defconfig](https://github.com/zephyrproject-rtos/zephyr/pull/81308/commits/f2b55b00f36484e9452e2f5481346898a51e4563#diff-044af1612b89d9e03c0c285d11e2e5946a1c97bc2bc6a47a0349b3b8e52019c4) under the `subsys/usb` then I am fine with `REQUIRES_SERIAL_BACKEND_CDC_ACM`, also it might lead to a better integration.
> 
> why would that file need to be placed under `subsys/usb` ? The symbols are defined here:

The symbols could be placed there as well. The thing is, snippets cannot contain code, and snippets do not support kconfig files. Boards like the nrf52840 dongle should not use snippets for the CDC ACM serial backend, but have it by default. This means that to switch to the new device support, I need to [add application code](https://github.com/zephyrproject-rtos/zephyr/pull/80985/commits/acbb65d03d9f9c4f0a889f98687c452e6f4bffee) for CDC ACM serial support to subsys/usb. Then I can also put those kconfig options and defaults there.

> I'm not sure starting to allow `defconfig` files under various subsys / lib / etc. folders is a wise idea. That will make it even harder to follow from where defconfig files are picked up.

But this has been done for a long time.

Reply:
> > I'm not sure starting to allow defconfig files under various subsys / lib / etc. folders is a wise idea. That will make it even harder to follow from where defconfig files are picked up.

> But this has been done for a long time.

What do you mean has been done for a long time ?
Today we only allow defconfig files under `boards` and `socs`, not under lib / subsys.